packages/jellyfish-api-core/__tests__/category/wallet/listWallets.test.ts

```diff
@@ -0,0 +1,39 @@
+import { Testing } from '@defichain/jellyfish-testing'
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+describe('Wallet without masternode', () => {
+  const container = new RegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await client.wallet.createWallet('test')
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should listWallets', async () => {
+    const wallets = await client.wallet.listWallets()
+    expect(wallets).toStrictEqual(['', 'test'])
+  })
+})
+
+describe('Wallet with masternode', () => {
+  const { container, rpc: { wallet } } = Testing.create(new MasterNodeRegTestContainer())
+
+  beforeAll(async () => {
+    await container.start()
+    await wallet.createWallet('test')
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should listWallets', async () => {
+    await expect(wallet.listWallets()).resolves.toEqual(['', 'test'])
```

Comment:
Same for this

Reply:
done, what is your thoughts on https://github.com/DeFiCh/jellyfish/pull/953/files#diff-1045b44753a823e6c6779b841a5e8d54fb6286c2961837d95fb2232b711b7218R25 ?

Would you expect something more like

```ts
const master = new MasterNodeRegTestContainer()
const { container, rpc: { wallet } } = Testing.create(master)
```

or

```ts
const master = new MasterNodeRegTestContainer()
const container = Testing.create(master)
```

Personally I am more leaning towards the way its currently written, but would be good to align with expectations in Jellyfish for these sort of approaches.

Pulling out variables of objects like this was a pretty neat ES update and I find myself using it quite often in other projects for method arguments and variable assignment.

[Destructuring Assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

Reply:
IMO, destructuring is great in functional method parameters, but **in tests**, it's pretty counterintuitive as it introduces too much symbol memorization when one is trying to understand what is going on.

I am also towards reducing unnecessary symbols that are not part of the testing context. By only introducing symbols that are part of the testing context. Optimizing for test readability over anything else, as for writing test and writing component we should take a different approach.

E.g.
```ts
const master = new MasterNodeRegTestContainer()
const { container, rpc: { wallet } } = Testing.create(master)
```

For the above, you expose 2 new symbols for the reader to track and understand as they are reading the test.

Vs.
```ts
const testing = Testing.create(new MasterNodeRegTestContainer())
```

For the above, you only have 1 symbol which is the "`testing`" framework object for the reader to follow along.

Subsequently:

```ts
testing.container.generate(1)
testing.rpc.wallet.doSomething(1)
```

It's contextually more understandable for the user to know that you are accessing the testing - container - to generate 1 block.

That's for tests. And it only makes sense **for tests**, as you are trying to make it as effortless as possible for the reviewer to understand what's going on. **However for non-tests**, I am in favor of destructuring as it reduces non-relevant symbols by focusing on what is required for each method/function.

Reply:
Great, thanks for the feedback and I agree with your views on this. Will update the tests to reflect this pattern.

Reply:
Updated tests, thanks again for the input on this 👍🏻